lib/modules/solidity/solcP.js

```diff
@@ -24,34 +32,59 @@ class SolcProcess extends ProcessWrapper {
     return {error: 'File not found'};
   }
 
-  loadCompiler(solcLocation) {
-    this.solc = require(solcLocation);
+  installAndLoadCompiler(solcVersion, packagePath){
+    let self = this;
+    return new Promise((resolve) => {
+      let manager = new PluginManager({pluginsPath: packagePath});
+      let timer;
+      if (!fs.existsSync(packagePath)) {
+        timer = new NpmTimer({logger: self._logger, packageName: 'solc', version: solcVersion, showSpinner: true});
+      }
+
+      if(timer) timer.start();
+      manager.install('solc', solcVersion).then(() => {
+        self.solc = manager.require('solc');
+        if(timer) timer.end();
+        resolve();
+      });
```

Comment:
There must be a `catch` that can be done, in case installation fails.

Reply:
Fixed.